lms/models/region.py

```diff
@@ -16,39 +16,3 @@ def __post_init__(self):
 
         #: Human readable description of the region.
         self.name: str = names[self.code]
-
-
-class Regions:
-    """A collection of all the regions as an enum like object."""
```

Comment:
This docstring was no longer accurate since https://github.com/hypothesis/lms/pull/5993 removed the hard-coded `Region` instances. The `Regions` class had become just an implementation of a global singleton (the `Regions._current_region` class attribute).

There's no need for this to be a singleton: it's not expensive to construct a `Region` object. We don't need to cache it.

It seems to have been done just to get around the fact that model code needs to access the "current" region but does not have access to the Pyramid settings (`region_code` and `authority`) that're needed to construct it. So this mechanism was set up whereby `app.py::create_app()` passes the settings into `Regions.set_region()` which stores them in a singleton which `Regions.get_region()` later returns when called by other model code. 

If model code really can't avoid needing access to the `region_code` and `authority` settings then it's more straightforward just to read them from either `os.environ` or [`pyramid.threadlocal.get_current_registry()`](https://docs.pylonsproject.org/projects/pyramid/en/latest/api/threadlocal.html#pyramid.threadlocal.get_current_registry)